hilti/include/rt/types/vector.h

```diff
@@ -251,15 +404,56 @@ inline std::ostream& operator<<(std::ostream& out, const vector::Empty& x) {
 }
 
 template<typename T, typename Allocator>
-inline std::ostream& operator<<(std::ostream& out, const vector::SafeIterator<T, Allocator>& x) {
-    out << to_string(x);
-    return out;
+bool operator!=(const Vector<T, Allocator>& a, const Vector<T, Allocator>& b) {
+    return ! (a == b);
 }
 
 template<typename T, typename Allocator>
-inline std::ostream& operator<<(std::ostream& out, const vector::SafeConstIterator<T, Allocator>& x) {
-    out << to_string(x);
-    return out;
+typename vector::Iterator<T, Allocator>::reference vector::Iterator<T, Allocator>::operator*() {
+    if ( auto&& c = _container(); c ) {
```

Comment:
Nit: why the `; c` )?

Reply:
Urg, that's horrible. Looks like brain damage from looking at Golang code. I'll fix it.